tests/e2e/metricschecker/metricschecker.go

```diff
@@ -0,0 +1,218 @@
+//go:build !windows
+
+package metricschecker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	clmo "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	"github.com/cilium/tetragon/tests/e2e/state"
+)
+
+// MetricsChecker checks prometheus metrics from one or more events streams.
+type MetricsChecker struct {
+	name      string
+	timeLimit time.Duration
+}
+
+// NewMetricsChecker constructs a new Metrics from a MultiEventChecker.
+func NewMetricsChecker(name string) *MetricsChecker {
+	rc := &MetricsChecker{
+		name:      name,
+		timeLimit: 30 * time.Second,
+	}
+	return rc
+}
+
+// WithTimeLimit sets the time limit for a MetricsChecker http GET calls.
+func (rc *MetricsChecker) WithTimeLimit(limit time.Duration) *MetricsChecker {
+	rc.timeLimit = limit
+	return rc
+}
+
+type metricOp int
+
+const (
+	opEqual   metricOp = 1
+	opGreater          = 1 << iota
+	opLess
```

Comment:
I guess we are using bits to capture things like `>=` as `opEqual | opLess`? Maybe add a comment here since it's  unconventional? (at least to me)

Reply:
Yep i'll do! :)

Reply:
Renamed so that `bitmask` usage is more obvious.